pkg/descheduler/strategies/failedpods.go

```diff
@@ -143,6 +144,10 @@ func validateFailedPodShouldEvict(pod *v1.Pod, strategyParams validatedFailedPod
 	if len(strategyParams.reasons) > 0 {
 		reasons := getFailedContainerStatusReasons(pod.Status.ContainerStatuses)
 
+		if pod.Status.Reason != "" {
```

Comment:
With `pod.Status.Phase == v1.PodFailed` as well just in case `validateFailedPodShouldEvict` gives more than just failed pods. Even though `ListPodsOnANodeWithFieldSelector` with the `status.phase=Failed` field selector is currently present, the future refactoring may break the assumption. The check is very simple.

Reply:
Applied, thank you.